weedcoco/importers/deposit.py

```diff
@@ -0,0 +1,107 @@
+import argparse
+import pathlib
+import json
+import os
+import re
+from shutil import copyfile
+from functools import reduce
+from weedcoco.utils import get_image_average_hash
+from weedcoco.validation import ValidationError, validate
+
+
+def validate_duplicate_images(image_hash):
+    if len(set(image_hash.values())) != len(image_hash):
+        raise ValidationError("There are identical images in the image directory.")
+
+
+def validate_existing_images(repository_dir, image_hash):
+    addition_hash = set(image_hash.values())
+    all_existing_hash = get_all_existing_hash(repository_dir)
+    if len(all_existing_hash.union(addition_hash)) != len(all_existing_hash) + len(
+        addition_hash
+    ):
+        raise ValidationError("There are identical images in the repository.")
+
+
+def get_all_existing_hash(repository_dir):
+    return reduce(
+        lambda acc, it: acc.union(set(it[2])),
+        filter(lambda dir: dir[0].endswith("/images"), os.walk(repository_dir)),
+        set(),
+    )
```

Comment:
I personally find following much more readable:
```suggestion
    return {
        filename
        for dirpath, dirnames, filenames
        if dirpath.endswith("/images")
        for filename in filenames
    }
```
I note two things about `.endswith("/images")`:
1. This won't work on Windows. Using `basename()` would be more cross-platform.
2. If we use a directory structure (e.g. using the first two characters of the image hash as a directory name) to split up large directories of images, then you'll need a different filter.

Reply:
I think this function has forgotten to remove the filename extension.

Reply:
```python
return {
        os.path.splitext(filename)[0]
        for dirpath, dirnames, filenames in os.walk(repository_dir)
        if os.path.basename(dirpath) == "images"
        for filename in filenames
    }
```

Reply:
Looks more precise, and readable